packages/firebase_auth/firebase_auth_web/lib/src/utils/web_utils.dart

```diff
@@ -22,20 +37,36 @@ FirebaseAuthException getFirebaseAuthException(
   Object exception, [
   auth_interop.Auth? auth,
 ]) {
-  if (exception is! core_interop.FirebaseError) {
+  if (!_isFirebaseAuthError(exception)) {
     return FirebaseAuthException(
       code: 'unknown',
       message: 'An unknown error occurred: $exception',
     );
   }
 
-  auth_interop.AuthError firebaseError = exception as auth_interop.AuthError;
-
-  String code = firebaseError.code.replaceFirst('auth/', '');
-  String message = firebaseError.message
-      .replaceFirst(' (${firebaseError.code}).', '')
+  String code = getProperty(exception, 'code').replaceFirst('auth/', '');
```

Comment:
Consider keeping the old pattern that casts to AuthError and then uses regular properties in Dart code. Casts to static interop types like AuthError will not break due to changes in DDC/dart2js (unlike the cast to the non-static interop types).

The reason behind this recommendation is to leverage the declared API and avoid subtle issues with `js_util`. Technically there are no issues here, but we generally recommend limiting the use of the js_util methods, like `getProperty`, because the extension declarations are safer and less error prone.

Above we could technically avoid the `getProperty(globalThis, 'Error'))` by writing instead:
```dart
@JS('Error')
external Object get errorConstructor;

bool _isirebaseAuthError(Object e) {
  instanceof(e, errorConstructor) &&
  hasProperty(e, 'customData')
}
```

Reply:
Thanks for the suggestion, I have reverted the cast back to as it was, I didn't expect it to work but work it did.

I've kept the `_isFirebaseAuthError(exception)` function as it is, your suggestion caused the following exception:
```
TypeError: Right-hand side of 'instanceof' is not an object
dart-sdk/lib/_internal/js_shared/lib/js_util_patch.dart 231:35       $instanceof
packages/firebase_auth_web/src/utils/web_utils.dart 29:50            _isFirebaseAuthError
```

Reply:
Thanks!

> I have reverted the cast back to as it was, I didn't expect it to work but work it did.

Glad to hear :) - yes, with static-interop, it all works better. This is about to improve further with new efforts in improving how we write interop code.

> I've kept the _isFirebaseAuthError(exception) function as it is, your suggestion caused the following exception

Sounds good to keep as you did. I'm curious about the error, though. I didn't see it when I tried locally, so there may be a small difference in what we both did.

One possible reason is where we both placed the definition of `errorConstructor`. If by chance you added it in a library like `auth_interop.dart`, then because the library has a `@JS` annotation itself, it changes the meaning of what the getter does. In a library with no namespace, the external member reads `window.Error`, but in a library like auth_interop, it would read `window.firebase_auth.Error`.

Could that be why?